src/authplus.ts

```diff
@@ -20,4 +28,37 @@ export class AuthPlus extends GoogleAuth {
   Compute = Compute;
   // tslint:disable-next-line: variable-name
   OAuth2 = OAuth2Client;
+  // tslint:disable-next-line: variable-name
+  GoogleAuth = GoogleAuth;
+
+  private _cachedAuth: GoogleAuth | null = null;
+
+  /**
+   * Override getClient(), memoizing an instance of auth for
+   * subsequent calls to getProjectId().
+   */
+  async getClient(
```

Comment:
Isn't this exactly the kind of thing we want to avoid?  Is this a method we would expect end users to consume?

Reply:
I don't _want_ people to use this method, but I think it's a bit too sweeping to remove the method at this time. Here's why I think this implementation, for the benefit of googlepapis, is a bit better:

1. if I call `getClient()` multiple times, the prior call will not have an effect on the values returned by the second call (because a `new GoogleAuth()` is created in between).
2. if I call `getClient(opts); getProjectId(); getClient(); getProjectId()` the cached response from the first `getClient` won't have an effect on the second `getProjectId()`.